File: Python/Problem_242.py
from collections import Counter

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return Counter(s) == Counter(t)

if __name__ == "__main__":
    sol = Solution()

    s1 = "anagram"
    t1 = "nagaram"
    s2 = "rat"
    t2 = "car"
    s3 = "aa"
    t3 = "bb"

    print(sol.isAnagram(s1, t1))
    print(sol.isAnagram(s2, t2))
    print(sol.isAnagram(s3, t3))

File: Cpp/Problem_242.cpp
#include <iostream>
#include <string>
#include <vector>

using namespace std;

class Solution {
public:
	bool isAnagram(string s, string t) {
		if (s.size() != t.size()) {
			return false;
		}

		vector <int> hash(26, 0);
		int result = 0;
		for (int i = 0; i < s.size(); i++) {
			++hash[s[i] - 'a'] <= 0 ? --result : ++result;
			--hash[t[i] - 'a'] >= 0 ? --result : ++result;
		}

		return !result;
	}
};

int main() {

	Solution sol;

	string s1 = "anagram", t1 = "nagaram";
	string s2 = "rat", t2 = "car";
	string s3 = "aa", t3 = "bb";

	cout << sol.isAnagram(s1, t1) << endl;
	cout << sol.isAnagram(s2, t2) << endl;
	cout << sol.isAnagram(s3, t3) << endl;

	return 0;
}